### PR TITLE
refactor: small fixes and refactoring

### PR DESCRIPTION
**Pull request by jorainer**
Smaller refactorings.

Thinking a little ahead: `spectraData` is expected to return missing values for spectra variables that are not provided by the original file. For that it might be maybe helpful to check the code in the `MsBackendCached` backend ([here](https://github.com/rformassspectrometry/Spectra/blob/master/R/MsBackendCached.R#L311-L351))? Basically, for all requested core spectra variables that are not available in `@frames` or `tms@all_columns` do the same as in that function: return `NA` of the correct data type.

### CODE REVIEW

File: R/MsBackendTimsTof-functions.R [hunk 1]
```diff
@@ -12,20 +12,25 @@
...
-  x@indices <- do.call(rbind, lapply(L, "[[", 2))
-  x@fileNames <- file
-  x
+    L <- bplapply(seq_len(length(file)), function(fl_idx) {
+        tms <- opentimsr::OpenTIMS(file[fl_idx])
+        on.exit(opentimsr::CloseTIMS(tms))
+        frames <- cbind(tms@frames, file = fl_idx)
+        indices <- cbind(
+            frame = rep(tms@frames$Id, tms@frames$NumScans),
+            scan = sequence(tms@frames$NumScans),
```
Comment (jorainer) [inline R/MsBackendTimsTof-functions.R:21]: just found this function. It's extremely fast.

File: R/MsBackendTimsTof-functions.R [hunk 2]
```diff
@@ -12,20 +12,25 @@
...
+            file = fl_idx)
+        list(frames, indices)
+    }, BPPARAM = BPPARAM)
+    x@frames <- do.call(rbindFill, lapply(L, "[[", 1))
+    idx <- match(colnames(x@frames, .SPECTRA_VARIABLE_MAPPINGS))
```
Comment (jorainer) [inline R/MsBackendTimsTof-functions.R:26]: the idea here is to rename some columns in `@frames` that it matches the corresponding spectra variables

File: R/MsBackendTimsTof-functions.R [hunk 3]
```diff
@@ -12,20 +12,25 @@
...
+    not_na <- !is.na(idx)
+    colnames(x@frames)[not_na] <- names(.SPECTRA_VARIABLE_MAPPINGS)[idx[not_na]]
+    if (any(colnames(x@frames) == "polarity"))
```
Comment (jorainer) [inline R/MsBackendTimsTof-functions.R:29]: maybe we need to do this also for other variables. Basically, to ensure the correct data type/format.

File: R/MsBackendTimsTof-functions.R [hunk 4]
```diff
@@ -135,4 +143,37 @@ MsBackendTimsTof <- function() {
...
+#' care of eventually duplicating values.
+#'
+#' @param x `MsBackendTimsTOF`
+#'
+#' @param columns `character` with the column names.
+#'
+#' @author Andrea Vicini, Johannes Rainer
+#'
+#' @noRd
+.get_frame_columns <- function(x, columns) {
```
Comment (jorainer) [inline R/MsBackendTimsTof-functions.R:156]: this is to avoid code duplication for all variables/columns that we want to extract from the `@frames`

**Review by andreavicini — APPROVED** [review]
Everything seem fine to me! Thanks @jorainer !

**Pull request merged**